File: Java/avion.java
/*
 *	Problem: Avion
 */

import java.util.Scanner;

public class avion {
	
	/*-----tracker-----*/
	public static void tracker(Scanner sc) {
		String output = "";
		
		for (int i = 1; i <= 5; i++) {
			String input = sc.next();
			
			if ( input.contains("FBI") )
				output += i + " ";
		}
		
		if ( output.equals("") )
			output = "HE GOT AWAY!";
		else
			output = output.substring(0, (output.length()-1) );
		
		System.out.println(output);
	} // end tracker
	
	/*-----main-----*/
	public static void main(String[] args) {		
		Scanner sc = new Scanner(System.in);
		tracker(sc);
	} //end main
	
} // end class

File: Ruby/avion.rb
=begin
 *
 *	Problem:		Avion
 *	Problem ID:	avion
 *	url:				https://open.kattis.com/problems/avion
 *	Solution:		Ruby
 *
=end

$number_cases = 5

class Solver
	
	def solve		
		blimp = false
		output = ""
		
		for case_number in 1..$number_cases
			current_code = gets.chomp
			if (current_code.include? "FBI")
				blimp = true
				output += case_number.to_s + " "
			end
		end
		
		if (blimp)
			puts output
		else
			puts "HE GOT AWAY!"
		end		
	end
	
end

solver = Solver.new
solver.solve
